src/analysis/convex/basic.lean

```diff
@@ -444,6 +444,33 @@ lemma convex.sub {t : set E}  (hs : convex s) (ht : convex t) :
   convex ((λx : E × E, x.1 - x.2) '' (s.prod t)) :=
 (hs.prod ht).is_linear_image is_linear_map.is_linear_map_sub
 
+lemma convex.smul_add_smul (h_conv : convex s) {p q : ℝ} (hple : 0 ≤ p) (hqle : 0 ≤ q) :
+  p • s + q • s = (p + q) • s :=
```

Comment:
What are your thoughts on changing this to
```suggestion
lemma convex.add_smul (h_conv : convex s) {p q : ℝ} (hple : 0 ≤ p) (hqle : 0 ≤ q) :
  (p + q) • s = p • s + q • s :=
```
to match `_root_.add_smul`? I assume the proof might need to change very slightly, but obviously not in a difficult way.

Reply:
Just doing the symmetric version I don't mind either way, certainly in my intended application the rw was the way I have it here 1/2 S + 1/2 S = S, but I don't know which is likely simpler for smul of sets.

Reply:
I think it's probably worth flipping in that case - it's easy to add an `\l` for your use case, and the lemma should be easier to find if it matches the name of an existing lemma.

Reply:
ok done